Make it possible to run firefox with KDE filedialogs.

The call for parse_app_info_from_flatpak_info uses the /proc/$PID/root
file to access the flatpak information file.

We neither have support for /proc/$PID/root nor flatpaks or snaps.

With this fix and deskutils/plasma5-xdg-desktop-portal-kde it is now possible
to run firefox using the KDE file dialogs.

> GTK_USE_PORTAL=1 firefox

--- src/xdp-utils.c.patched	2021-07-22 16:25:27.756116000 +0200
+++ src/xdp-utils.c	2021-07-22 16:30:51.814245000 +0200
@@ -630,6 +630,11 @@
   g_autoptr(XdpAppInfo) app_info = NULL;
   g_autoptr(GError) local_error = NULL;
 
+// On FreeBSD the main use of the xdg-portal is to be able to use
+// "native" file dialogs, say the KDE file dialogs in firefox.
+// There is no support for the application package mechanisms like
+// flatpak and snap yet.
+#ifndef __FreeBSD__
   app_info = parse_app_info_from_flatpak_info (pid, &local_error);
   if (app_info == NULL && local_error)
     {
@@ -646,6 +651,7 @@
           return NULL;
         }
     }
+#endif
 
   if (app_info == NULL)
     app_info = xdp_app_info_new_host ();
